### PR TITLE
Write sub timings into yaml files

### PR DESCRIPTION
**Pull request by AtreyeeS**
Hello @cdeil !

I am facing some issue while writing the yaml files. The `data` dictionary that I create contains the timings for each process (as I see from the print statement). However, the `subtimings.yaml` (https://github.com/gammapy/gammapy-benchmarks/blob/851a113e353043b1b143494cb2d18cccb12505b4/benchmarks/results/analysis_3d/subtimings.yaml) file does not write float values for the timings.

How do I write a sensible output? Please advise

### CODE REVIEW

**Comment by cdeil** [comment]
YAML by default doesn't support Numpy or Astropy objects.

You have to call `float(x)` or `x.tolist()` if you have a scalar or array Numpy array.
Then the YAML output will be OK.

If you can't get it to work, let me know and I'll do the edits.

File: benchmarks/analysis_3d.py [hunk 1]
```diff
@@ -66,13 +86,55 @@ def run_benchmark():
...
 
+
+def flux_point(stacked):
     e_edges = [0.3, 1, 3, 10] * u.TeV
     fpe = FluxPointsEstimator(datasets=[stacked], e_edges=e_edges, source="gc-source")
     fpe.run()
 
 
+def run_benchmark():
+    times = []
```
Comment (cdeil) [inline benchmarks/analysis_3d.py]: Make an empty dict here, and then after each function, add an entry.
(never use a Python list or Numpy array of time points)

Comment (AtreyeeS) [reply benchmarks/analysis_3d.py]: Thanks @cdeil ! This works now. I will adapt the rest of the files and finish this PR

**Comment by AtreyeeS** [comment]
Adapted 1D and 3D analysis.

**Pull request merged**